### PR TITLE
Don't internalize when ILMerging and test the ILMerged nuget.exe

### PR DESCRIPTION
**Pull request by joelverhagen**
- Fixes https://github.com/NuGet/Home/issues/3753.
- No longer internalize types in nuget.exe. We now use `/allowDup:TYPE` deal with the duplicate types in NuGet.Frameworks.
- Tests ILMerged nuget.exe on the CI. Local builds test the unmerged version to allow easier debugging.

/cc @alpaix @jainaashish @emgarten

### CODE REVIEW

**Comment by alpaix** [comment]
I'd suggest having dedicated msbuild parameter smth. like `$(TestTargetDir)` initialized with `$(TargetDir)` by default in .props file (instead of `$(ILMergeOutputDir)`). This would make it more generic. For instance, one would be able to run tests with arbitrary NuGet.exe not necessarily ilmerge'd version.
Plus change in `GetNugetExe` implementation would be minimal (thus would make it easier for the escrow.

Does it make sense?

File: src/NuGet.Clients/NuGet.CommandLine/.mergeallowdup [hunk 1]
```diff
@@ -0,0 +1,35 @@
+NuGet.Frameworks.CompatibilityListProvider
```
Comment (alpaix) [inline src/NuGet.Clients/NuGet.CommandLine/.mergeallowdup:1]: Can you please sort this list?

**Review by emgarten — APPROVED** [review]

**Review by jainaashish — APPROVED** [review]
Looks good

File: src/NuGet.Clients/NuGet.CommandLine/.mergeallowdup (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Clients/NuGet.CommandLine/.mergeallowdup:1]: Sure

**Comment by joelverhagen** [comment]
> Does it make sense?

Yes, will do.

**Review by alpaix — APPROVED** [review]
👏

**Pull request merged**